packages/engine.io-client/lib/socket.ts

```diff
@@ -708,6 +715,32 @@ export class SocketWithoutUpgrade extends Emitter<
     return this.writeBuffer;
   }
 
+  /**
+   * Returns `true` if the connection is responding to heartbeats.
+   *
+   * If heartbeats are disabled this will always return `true`.
+   *
+   * @return {boolean}
+   */
+  public isResponsive() {
```

Comment:
To be honest, I'm not sure about this public method. This is something we'll have to maintain forever, and which is not part of the WebSocket API which we try to mimic (more or less). I'm not sure how we could implement it differently though...

Reply:
Looks like things like `_packet` are fake private? Could prefix it with an underscore and document it to be an internal api?

another potential idea was to add a return value to `_packet` which signals if the packet was actually written